lib/actions/ssh.ts

```diff
@@ -0,0 +1,182 @@
+support) please check the README here: https://github.com/resin-io/resin-cli
+
+Use this command to get a shell into the running application container of
+your device.
+
+Examples:
+
+	$ resin ssh MyApp
+	$ resin ssh 7cf02a6
+	$ resin ssh 7cf02a6 --port 8080
+	$ resin ssh 7cf02a6 -v
+	$ resin ssh 7cf02a6 -s\
+`,
+	permission: 'user',
+	primary: true,
+	options: [
+		{
+			signature: 'port',
+			parameter: 'port',
+			description: 'ssh gateway port',
+			alias: 'p',
+		},
+		{
+			signature: 'verbose',
+			boolean: true,
+			description: 'increase verbosity',
+			alias: 'v',
+		},
+		commandOptions.hostOSAccess,
+		{
+			signature: 'noproxy',
+			boolean: true,
+			description: `don't use the proxy configuration for this connection. \
+Only makes sense if you've configured proxy globally.`,
+		},
+	],
+	async action(params, options, done) {
+		const childProcess = await import('child_process');
+		const Bluebird = await import('bluebird');
+		const resin = (await import('resin-sdk')).fromSharedOptions();
+		const _ = await import('lodash');
+		const bash = await import('bash');
+		const hasbin = await import('hasbin');
+
+		// hasbin ignores the error-first convention, so we can't promisify automatically
+		const hasbinAsync = (bin: string) =>
+			new Promise<boolean>(resolve => hasbin(bin, resolve));
+
+		const { getSubShellCommand } = await import('../utils/helpers');
+		const patterns = await import('../utils/patterns');
+
+		if (options.port == null) {
+			options.port = 22;
+		}
+
+		const verbose = options.verbose ? '-vvv' : '';
+
+		const proxyConfig = global.PROXY_CONFIG;
+		const useProxy = !!proxyConfig && !options.noproxy;
+
+		const getSshProxyCommand = function(hasTunnelBin: boolean) {
+			if (!useProxy) {
+				return '';
+			}
+
+			if (!hasTunnelBin) {
+				console.warn(`\
+Proxy is enabled but the \`proxytunnel\` binary cannot be found.
+Please install it if you want to route the \`resin ssh\` requests through the proxy.
+Alternatively you can pass \`--noproxy\` param to the \`resin ssh\` command to ignore the proxy config
+for the \`ssh\` requests.
+
+Attempting the unproxied request for now.\
+`);
+				return '';
+			}
+
+			const tunnelOptions = {
+				proxy: `${proxyConfig!.host}:${proxyConfig!.port}`,
+				dest: '%h:%p',
+			};
+			const { proxyAuth } = proxyConfig!;
+			if (proxyAuth) {
+				const i = proxyAuth.indexOf(':');
+				_.assign(tunnelOptions, {
+					user: proxyAuth.substring(0, i),
+					pass: proxyAuth.substring(i + 1),
+				});
+			}
+			const proxyCommand = `proxytunnel ${bash.args(tunnelOptions, '--', '=')}`;
+			return `-o ${bash.args({ ProxyCommand: proxyCommand }, '', '=')}`;
+		};
+
+		const uuidExists =
+			params.uuid && (await resin.models.device.has(params.uuid));
+		const selectedUuid = uuidExists
+			? params.uuid!
+			: await patterns.inferOrSelectDevice();
+
+		console.info(`Connecting to: ${selectedUuid}`);
+		const device = await resin.models.device.get(selectedUuid, {});
```

Comment:
I'm not sure whether the empty options param is bc the sdk wasn't marking the param as optional, but if that's the case then this is a reminder to remove when the sdk is updated.